ledger/src/blockstore_db.rs

```diff
@@ -56,6 +60,12 @@ const DEFAULT_FIFO_COMPACTION_DATA_CF_SIZE: u64 = 125 * 1024 * 1024 * 1024;
 // used by coding shreds.
 const DEFAULT_FIFO_COMPACTION_CODING_CF_SIZE: u64 = 100 * 1024 * 1024 * 1024;
 
+// Thread local instance of RocksDB's PerfContext.
+thread_local! {static PER_THREAD_ROCKS_PERF_CONTEXT: RefCell<PerfContext> = RefCell::new(PerfContext::default());}
+
+// The sample rate of RocksDB's PerfContext in 1 out of N.
+const ROCKSDB_PERF_CONTEXT_SAMPLE_RATE: u32 = 100;
```

Comment:
For local tests or experimental runs, please feel free to change this number to 1 to enable full sample.

Reply:
nits:
- Maybe change "RATE" to "INTERVAL"; interval isn't a 100% match, but sample rate makes me think that a bigger number would result in more samples. Interval implies that a larger number will be fewer samples

Do you anticipate us wanting to change this number frequently? Should we provide a way to adjust this value (env variable or arg) without having to rebuild ?

Reply:
> Maybe change "RATE" to "INTERVAL"; interval isn't a 100% match, but sample rate makes me think that a bigger number would result in more samples. Interval implies that a larger number will be fewer samples

Maybe I can change it to something like (N out of 10000) or something that might be more intuitive?

> Do you anticipate us wanting to change this number frequently? Should we provide a way to adjust this value (env variable or arg) without having to rebuild ?

I've been thinking about that as well.  In normal situations, we probably don't change it unless we are debugging something.  Having an env arg with a default value will help in this case.

Reply:
>Maybe I can change it to something like (N out of 10000) or something that might be more intuitive?

One other idea might be to move the `1000` constant out of the `maybe_collect_perf_context()` into a second constant right above/below this one. Seeing that second constant right there could help, especially if the comment mentions the constant names explicitly.

> Having an env arg with a default value will help in this case.

Yep, this is where my mind wandered too.